### PR TITLE
Added states for installing and configuring reddit instance

### PR DESCRIPTION
**Pull request by blarghmatey**

### CODE REVIEW

File: reddit/install.sls [hunk 1]
```diff
@@ -3,8 +3,115 @@
...
+        stop on runlevel [!2345] or reddit-restart all or reddit-restart websockets
+        start on runlevel [2345] or reddit-restart all or reddit-restart websockets
+
+        respawn
+        respawn limit 10 5
+        kill timeout 15
+
+        limit nofile 65535 65535
+
+        exec baseplate-serve2 --bind localhost:9001 $REDDIT_SRC/websockets/example.ini
```
Comment (shaidar) [inline reddit/install.sls]: Should this be `example.ini`?

File: reddit/mcrouter.sls [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+create_mcrouter_configuration:
+  file.managed:
+    - name: /etc/mcrouter/global.conf
+    - makedirs: True
+    - contents: |
+        {{ reddit.mcrouter_config|json(indent=2)|indent(8) }}
+
+create_mcrouter_default_setting:
+  file.managed:
+    - name: /etc/default/mcrouter
```
Comment (shaidar) [inline reddit/mcrouter.sls:12]: This file has no extension?

File: reddit/config.sls [hunk 3]
```diff
@@ -1,15 +1,63 @@
 {% from "reddit/map.jinja" import reddit with context %}
 
-include:
-  - .install
-  - .service
+{% for conf_name, val in reddit.queu_config_count.items() %}
```
Comment (shaidar) [inline reddit/config.sls]: Typo `queu`

File: reddit/post_install.sls [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+          create or replace function domain(url text) returns text as $$
+          select substring($1 from E'(?i)(?:.+?://)?(?:www[\d]*\.)?([^#/]*)/?')
+          $$ language sql immutable;
+      - host: postgresql.service.consul
+      - user: {{ pg_user }}
+      - password: {{ pg_pass }}
+      - write: True
+      - maintenance_db: reddit
+
+restart_all_redit_services:
```
Comment (shaidar) [inline reddit/post_install.sls]: Knit pick- `redit`

**Comment by shaidar** [comment]
:thumbsup:

File: reddit/mcrouter.sls (hunk 2 above)
Comment (blarghmatey) [reply reddit/mcrouter.sls:12]: Yes, that is correct. The convention in the /etc/defaults folder is to not have a file extension.

**Pull request merged**